### DISCUSSION TITLE
API breaking change cart checkout_url value

### DISCUSSION THREAD

**Discussion by vettloffah** [swellstores/community#202] [Help] [answered]
Since swell fixed their frontend API outage today, the value of the cart `checkout_url` property has been coming in differently than before, using our main site domain instead of the `swell.store` domain, causing a 404 error for all users when trying to checkout.

I resolved the issue by manually creating the checkout link using the `https://<store_id>.swell.store/checkout/<checkout_id>` format.

Posting here since others are certainly experiencing the issue if using the hosted checkout. I'm really wondering how a serious breaking change like this makes it through QA and gets published. With this issue plus the API outage, we haven't processed a single transaction since last night.

**Comment by osseonews** [comment]
Agree 100%! The breaking change with checkout_url is happening to us also , and this probably explains why some API endpoints are still not working for us, e.g. The /recovery endpoint always returns an Error: Cart checkout is expired, recovery not allowed. But, we cannot figure out how to fix this. Can you please explain what you mean by manually creating the checkout to fix this error? You actually send your customer a different URL manually?

**Reply by vettloffah** [reply]
In my case when someone clicks the "start checkout" button on the car page, we forward them to the checkout url. 

So we have to return a url thats different then the cart.checkout_url since that's broken as of today. I posted the link format in the original post here.

**Comment by logeshswell** [comment, marked as answer]
@vettloffah @osseonews The fix for this bug is rolled out to production. Can you check and confirm?

**Comment by ericingram** [comment]
> I'm really wondering how a serious breaking change like this makes it through QA and gets published

We're looking into it, but essentially we made a major necessary change to Storefronts and obviously missed some tests. It's unfortunate and I apologize for the trouble on our behalf. We are working to improve testing practices.
